src/main/java/de/tum/in/www1/artemis/domain/scoring/ScoringStrategyShortAnswerProportionalWithPenalty.java

```diff
@@ -0,0 +1,81 @@
+package de.tum.in.www1.artemis.domain.scoring;
+
+import de.tum.in.www1.artemis.domain.*;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Proportional with Penalty means that
+ * every correct mapping increases the score by x and
+ * every incorrect mapping decreases the score by x
+ * where x = maxScore / numberOfDropLocationsThatShouldHaveAMapping
+ * if the result is negative, a score of 0 is given instead
+ */
+public class ScoringStrategyShortAnswerProportionalWithPenalty implements ScoringStrategy{
+    @Override
+    public double calculateScore(Question question, SubmittedAnswer submittedAnswer) {
+        //check if the question is invalid: if true: -> return with full points
+        if (question.isInvalid()) {
+            return question.getScore();
+        }
+
+        if (submittedAnswer instanceof ShortAnswerSubmittedAnswer && question instanceof ShortAnswerQuestion) {
+            ShortAnswerSubmittedAnswer saAnswer = (ShortAnswerSubmittedAnswer) submittedAnswer;
+            ShortAnswerQuestion saQuestion = (ShortAnswerQuestion) question;
+
+            boolean foundCorrectSolution;
+            double totalSolutions = saQuestion.getSpots().size();
+            double correctSolutions = 0;
+            double incorrectSolutions = 0;
+            Set<ShortAnswerSolution> solutionsAlreadyUsed = new HashSet<>();
+
+            // iterate through each spot and compare its correct solutions with the submitted texts
+            for (ShortAnswerSpot spot : saQuestion.getSpots()) {
```

Comment:
This code is duplicated, please either delegate to a Util class, or create an intermediate abstract super class called AbstractScoringStrategyShortAnswer

Reply:
Created an ScoringStrategyShortAnswerUtil class to remove the duplicated code.